### DISCUSSION TITLE
Embbed Properties are not written  into db on upsert

### DISCUSSION THREAD

**Discussion by Johanneslueke** [mikro-orm/mikro-orm#5795] [General]
### Describe the bug

I wanted to update an existing entity with values of an embbed property. the default value (null) is never updated on upsert.

previously the embbeded property didn't exist on the entity. so after adding it i created a migration and run it. i allow the values to be null, because previous entries didn't exist and i could not come up with proper default values.

Now i am stuck with the fields of the embbeded property  always being "null". 

According to the documentation upsert does the following: 

> This method performs an _insert on conflict merge_ query ensuring the database is in sync

Which i understand that if the values differ the entity in the db will be updated  with the new values. Am I wrong with that assumption?

The next curious thing is **IF the entity did not exist at all in the table and we run upsert, then the embbeded property is correctly written into to the Database table**

this issue seems similar or at least somwhat related to #5500 

### Reproduction

```ts

const client = { 
   id: "abc",
   ownerID: "def",
   name: "heinzDieter",
   type: "tcp",
   state: {
      serverTimestamp: "a time string as plain string",
      timestamp: "a time string as plain string",
      status: "ok"
   }
};


const ClientRepo = em.getRepository(Client); 

await ClientRepo.upsert({
             id: client.id,
             ownerID: client.ownerID,
             type: client.type,
             name: client.name,
             //state: client.state as ClientState, // not writting any values to get fields of value "null"
});

await ClientRepo.upsert({
             id: client.id,
             ownerID: client.ownerID,
             type: client.type,
             name: client.name,
             state: client.state as ClientState, // trying to override nulled fields, does fail -> fields in table stay null
});

```

```ts

@Embeddable()
export class ClientState {
  @Property()
  serverTimestamp!: string;

  @Property()
  status!: string;

  @Property()
  timestamp!: string;
}

@Entity({ repository: () => ClientRepository })
@Index({ properties: ['ownerID', 'name'] })
@Unique({ properties: ['ownerID', 'name'] })
export class Client {
  [EntityRepositoryType]?: ClientRepository;
  @PrimaryKey({ autoincrement: false, type: 'string' })
  id!: string;

  @Property({
    onCreate: () => new Date().toISOString(),
    name: 'created_at',
    nullable: false,
    default: new Date().toISOString(),
    comment: 'DateTime in ISO format',
  })
  createAt!: string;

  @Property({
    onUpdate: () => new Date().toISOString(),
    name: 'update_at',
    nullable: false,
    default: new Date().toISOString(),
    comment: 'DateTime in ISO format',
  })
  updatedAt!: string;

  @Property({})
  ownerID!: string;

  @Property()
  name!: string;

  @Embedded({
    persist: true,
    entity: () => ClientState,
    nullable: true,
  })
  state?: ClientState = new ClientState();


  @Enum({ items: () => ClientType })
  type!: ClientType;
}

export class ClientRepository extends EntityRepository<Client> {
}

```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

 @mikro-orm/cli@6.2.9 ├── @mikro-orm/core@6.2.9 ├── @mikro-orm/migrations@6.2.9 ├── @mikro-orm/nestjs@6.0.2 ├── @mikro-orm/postgresql@6.2.9 ├── @mikro-orm/reflection@6.2.9

### Node.js version

20.11

### Operating system

Ubunut 22.04

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
The second call will not do anything, since you try to upsert already managed entity by its primary key. You need to flush for the changes to propagate to the database. This is the code path that should be taken with your repro:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts#L888

**Reply by B4nan** [reply]
I can't say I am entirely happy with how this works now. Maybe we could add an option (as well as global config) to always fire the query and synchronize the state (and maybe even change the default behavior). You are not the first one confused by this.

**Reply by Johanneslueke** [reply]
Indeed this was confusing. But finding the managed element and assigning the change did the trick.

I would support your idea making the synchronization of the state a default behaviour. Or at least specifically document that the upsert method does not necessarily execute an actual update.

**Reply by B4nan** [reply]
This behavior is explicitly documented, both in the [JSDOC](https://mikro-orm.io/api/core/class/EntityManager#upsert) and in the [documentation](https://mikro-orm.io/docs/entity-manager#upsert).
